sdk/communication/azure-communication-rooms/README.md

```diff
@@ -0,0 +1,289 @@
+A `DefaultAzureCredential` object must be passed to the `RoomsClientBuilder` via the credential() function. Endpoint and httpClient must also be set via the endpoint() and httpClient() functions respectively.
+
+`AZURE_CLIENT_SECRET`, `AZURE_CLIENT_ID` and `AZURE_TENANT_ID` environment variables
+are needed to create a DefaultAzureCredential object.
+
+Alternatively, you can provide the entire connection string using the connectionString() function instead of providing the endpoint and access key.
+<!-- embedme src/samples/java/com/azure/communication/rooms/ReadmeSamples.java#L45-L53 -->
+```java
+
+public RoomsClient createRoomsClientWithConnectionString() {
+    // You can find your connection string from your resource in the Azure Portal
+    String connectionString = "https://<resource-name>.communication.azure.com/;<access-key>";
+
+    RoomsClient roomsClient = new RoomsClientBuilder().connectionString(connectionString).buildClient();
+
+    return roomsClient;
+}
+```
+
+## Key concepts
+
+There are four operations to interact with the Azure Communication Rooms Service.
+
+## Examples
+
+### Create a new room
+Use the `createRoom`  function to create a new Room on Azure Communciation Service.
+
+<!-- embedme src/samples/java/com/azure/communication/rooms/ReadmeSamples.java#L76-L87 -->
+```java
+public void createRoomWithValidInput() {
+    OffsetDateTime validFrom = OffsetDateTime.of(2021, 8, 1, 5, 30, 20, 10, ZoneOffset.UTC);
+    OffsetDateTime validUntil = OffsetDateTime.of(2021, 9, 1, 5, 30, 20, 10, ZoneOffset.UTC);
+    List<RoomParticipant> participants = new ArrayList<>();
+    // Add two participants
+    participants.add(new RoomParticipant("<ACS User MRI identity 1>", "Prebuilt Role Name"));
+    participants.add(new RoomParticipant("<ACS User MRI identity 2>", "Prebuilt Role Name"));
+
+    RoomsClient roomsClient = createRoomsClientWithConnectionString();
+    CommunicationRoom roomResult = roomsClient.createRoom(validFrom, validUntil, participants);
+    System.out.println("Room Id: " + roomResult.getRoomId());
+}
+```
+
+### Create a new open room
+Use the `createRoom`  function to create a new Open Room on Azure Communciation Service.
+<!-- embedme src/samples/java/com/azure/communication/rooms/ReadmeSamples.java#L89-L96 -->
+```java
+public void createOpenRoomWithValidInput() {
+    OffsetDateTime validFrom = OffsetDateTime.of(2021, 8, 1, 5, 30, 20, 10, ZoneOffset.UTC);
+    OffsetDateTime validUntil = OffsetDateTime.of(2021, 9, 1, 5, 30, 20, 10, ZoneOffset.UTC);
+
+    RoomsClient roomsClient = createRoomsClientWithConnectionString();
+    CommunicationRoom roomResult = roomsClient.createRoom(validFrom, validUntil, true, null);
+    System.out.println("Room Id: " + roomResult.getRoomId());
+}
+```
+
+### Update an existing room
+Use the `updateRoom`  function to create a new Room on Azure Communciation Service.
+
+<!-- embedme src/samples/java/com/azure/communication/rooms/ReadmeSamples.java#L98-L115 -->
```

Comment:
Use the [codesnippet maven plugin](https://github.com/Azure/azure-sdk-tools/tree/main/packages/java-packages/codesnippet-maven-plugin) instead to inject code into README and in JavaDocs.

Reply:
Addressed.